src/reporting/clients/MappingsClient.ts

```diff
@@ -0,0 +1,641 @@
+    const requestOptions: RequestInit = this.createRequest("DELETE", accessToken);
+    return this.fetchJSON<Response>(url, requestOptions);
+  }
+
+  public async copyMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mappingCopy: MappingCopy
+  ): Promise<Mapping> {
+    if (null != mappingCopy.mappingName && !this.isSimpleIdentifier(mappingCopy.mappingName)) {
+      throw new RequiredError(
+        "mappingName",
+        "Field mappingName of mappingCopy was invalid when calling copyMapping.",
+      );
+    }
+    if(this.isNullOrWhitespace(mappingCopy.targetIModelId)) {
+      throw new RequiredError(
+        "targetiModelId",
+        "Required field targetiModelId of mappingCopy was missing when calling copyMapping.",
+      );
+    }
+
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings/${encodeURIComponent(mappingId)}/copy`;
+    const requestOptions: RequestInit = this.createRequest("POST", accessToken, JSON.stringify(mappingCopy));
+    return (await this.fetchJSON<MappingSingle>(url, requestOptions)).mapping;
+  }
+
+  public async getGroups(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    top?: number
+  ): Promise<Group[]> {
+    const groups: Array<Group> = [];
+    const groupIterator = this.getGroupsIterator(accessToken, iModelId, mappingId, top);
+    for await(const group of groupIterator) {
+      groups.push(group);
+    }
+    return groups;
+  }
+
+  public getGroupsIterator(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    top?: number
+  ): EntityListIterator<Group> {
+    if(!this.topIsValid(top)) {
+      throw new RequiredError(
+        "top",
+        "Parameter top was outside of the valid range [1-1000] when calling getGroupsIterator."
+      );
+    }
+    let url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings/${encodeURIComponent(mappingId)}/groups`;
+    url += top ? `/?%24top=${top}` : "";
+    const request = this.createRequest("GET", accessToken);
+    return new EntityListIteratorImpl(async () => getEntityCollectionPage<Group>(
+      url,
+      async (nextUrl: string): Promise<Collection<Group>> => {
+        const response: GroupCollection = await this.fetchJSON<GroupCollection>(nextUrl, request);
+        return {
+          values: response.groups,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          _links: response._links,
+        };
+      }));
+  }
+
+  public async createGroup(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    group: GroupCreate
+  ): Promise<Group> {
+    if(!this.isSimpleIdentifier(group.groupName)) {
+      throw new RequiredError(
+        "groupName",
+        "Required field mappingName of group was null or undefined when calling createGroup.",
```

Comment:
`Required field mappingName of group was invalid when calling createGroup.`

Reply:
Error message fixed